zombies.js

```diff
@@ -178,7 +243,31 @@
  * @name eat
  * @param {Food} itemToEat  The food item to eat.
  */
-
+  eat(itemToEat){
+
+    var itemIndex = this.getPack().indexOf(itemToEat);
+      //if the food is food and if it is in the bag:
+    if(itemToEat instanceof Food && itemIndex > -1){
+      //if the item's health += the itemtoeat's energy is greater than the max health:
+      console.log("OMNOMNOMN...");  
+      if(this.health + itemToEat.energy > this.getMaxHealth() ){
+        //Then set the health equal to the max health and then discard from bag
+        this.health = this.getMaxHealth();
+        this.discardItem(itemToEat);
```

Comment:
There is an optimization that could be done here where the call to `this.discardItem(itemToEat)` only needs to be called once.